keygen/src/keygen.rs

```diff
@@ -561,8 +560,11 @@ fn main() -> Result<(), Box<dyn error::Error>> {
                 ),
 
         )
-        .get_matches();
+}
 
+fn main() -> Result<(), Box<dyn error::Error>> {
+    let default_num_threads = num_cpus::get().to_string();
+    let matches = app(&default_num_threads, solana_version::version!()).get_matches();
```

Comment:
When we will migrate to clap-v3, get_matches might panic in case of some common cli code looks for arg which doesn't exist. For this case, it might be good to use `try_get_matches` for clap-v3 or later